deployment/terraform/create.go

```diff
@@ -180,62 +182,74 @@ func (t *Terraform) Create(initData bool) error {
 
 func (t *Terraform) setupAppServers(extAgent *ssh.ExtAgent, uploadBinary bool, binaryPath string) {
 	for _, val := range t.output.Instances {
-		ip := val.PublicIP
-		sshc, err := extAgent.NewClient(ip)
+		err := t.setupAppServer(extAgent, val.PublicIP, uploadBinary, binaryPath, false, !t.output.HasJobServer())
 		if err != nil {
-			mlog.Error("error in getting ssh connection", mlog.String("ip", ip), mlog.Err(err))
-			continue
+			mlog.Error("error while setting up app server", mlog.Err(err))
 		}
-		func() {
-			defer func() {
-				err := sshc.Close()
-				if err != nil {
-					mlog.Error("error closing ssh connection", mlog.Err(err))
-				}
-			}()
-
-			// Upload files
-			batch := []uploadInfo{
-				{srcData: strings.TrimSpace(serverSysctlConfig), dstPath: "/etc/sysctl.conf"},
-				{srcData: strings.TrimSpace(serviceFile), dstPath: "/lib/systemd/system/mattermost.service"},
-				{srcData: strings.TrimPrefix(limitsConfig, "\n"), dstPath: "/etc/security/limits.conf"},
-			}
-			if err := uploadBatch(sshc, batch); err != nil {
-				mlog.Error("batch upload failed", mlog.Err(err))
-				return
-			}
+	}
 
-			cmd := "sudo systemctl daemon-reload && sudo service mattermost stop"
-			if out, err := sshc.RunCommand(cmd); err != nil {
-				mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.String("output", string(out)), mlog.Err(err))
-				return
-			}
+	for _, val := range t.output.JobServers {
+		err := t.setupAppServer(extAgent, val.PublicIP, uploadBinary, binaryPath, true, true)
```

Comment:
0/5, but might be better to refactor these into `setupAppServer` and `setupJobServer`, where both of these internally call `setupServer` which contains the common bits. But they handle the initialization like passing the serviceFileName directly and the jobServerEnabled part.

Reply:
Sounds good to me.

Reply:
Do we even need to pass this `jobServerEnabled` flag around? We can derive that at any point from any terraform method.